### PR TITLE
dev-cmd/bump-formula-pr: always use full git repo as tap remote

### PR DESCRIPTION
**Pull request by dawidd6**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Hi there folks!

I'm currently taking care of my GitHub Action [action-homebrew-bump-formula](https://github.com/dawidd6/action-homebrew-bump-formula) and wanted to move my test formulae from separate repository to the same repository where the action is stored.
I've run into an issue, because the repository is not named _properly_ (i.e. `$USER/homebrew-$TAP`), `brew bump-formula-pr` would try to access non-existent repository named for example `dawidd6/tap`, it depends how I cloned the tap of course. I've tracked it down to this changed line. Didn't test it on homebrew-core, but I think it should work the same as for my repository.

Steps to reproduce:
1. `brew tap dawidd6/tap https://github.com/dawidd6/action-homebrew-bump-formula.git`
2. `brew bump-formula-pr --no-audit --no-browse --no-fork --version=0.1.12 --force dawidd6/tap/test-formula-url`
3. Observe error "Not found".

### CODE REVIEW

**Comment by dawidd6** [comment]
So, I've blamed this line and found this PR https://github.com/Homebrew/brew/pull/15082 :smile:.
Converting to draft for now, will find a solution working for both cases.

**Comment by dawidd6** [comment]
Should be better now I think.

File: Library/Homebrew/tap.rb [hunk 1]
```diff
@@ -276,9 +276,10 @@ def remote
   def remote_repository
     return unless (remote = self.remote)
 
-    @remote_repository ||= remote.delete_prefix("https://github.com/")
-                                 .delete_prefix("git@github.com:")
+    @remote_repository ||= remote.delete_prefix("https://")
+                                 .delete_prefix("git@")
                                  .delete_suffix(".git")
+                                 .split(%r{[:\/]})[-2..].join("/")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb:282]: Can you clarify a bit more what this is doing? It's pretty hard to follow.

Comment (MikeMcQuaid) [reply Library/Homebrew/tap.rb:282]: Can you use one of the existing regexes e.g. `HOMEBREW_TAP_FORMULA_REGEX` or similar for this instead?

Comment (dawidd6) [reply Library/Homebrew/tap.rb:282]: None of the existing regexes in `tap_constants.rb` or anywhere else catch my eye. `HOMEBREW_TAP_FORMULA_REGEX` matches only a tap and formula in format `owner/repo/formula`. Here we need to extract the owner and repository name from strings like:
- `git@github.com:owner/repo`
- `https://github.com/owner/repo`

Also any custom remote should be supported (not breaking the PR I've linked above) like:
- `https://mirror.something.com/github/owner/repo.git`

Comment (MikeMcQuaid) [reply Library/Homebrew/tap.rb:282]: @dawidd6 Can you add a new regex in that format then with named captures? Much easier to follow.

Comment (dawidd6) [reply Library/Homebrew/tap.rb]: Done and seems to work as expected :slightly_smiling_face:.

**Review by MikeMcQuaid — APPROVED** [review]
Thanks again @dawidd6!
